esda/shape.py

```diff
@@ -0,0 +1,332 @@
+    """
+    offset = int(0)
+    start = points[0]
+    on_geom = 0
+    on_coord = 0
+    result = []
+    n_points = len(points)
+    while True:
+        if on_coord == (n_coords_per_geom[on_geom] - 1):
+            offset += on_coord
+            on_geom += 1
+            on_coord = 0
+            if on_geom == len(n_coords_per_geom):
+                break
+            else:
+                continue
+        left = points[offset + on_coord % (n_coords_per_geom[on_geom] - 1)]
+        center = points[offset + (on_coord + 1) % (n_coords_per_geom[on_geom] - 1)]
+        right = points[offset + (on_coord + 2) % (n_coords_per_geom[on_geom] - 1)]
+        a = left - center
+        b = right - center
+        angle = numpy.math.atan2(a[0] * b[1] - a[1] * b[0], numpy.dot(a, b))
+        result.append(angle)
+        on_coord += 1
+    return result
+
+
+# -------------------- INTERTIAL MEASURES -------------------- #
+
+
+def isoperimetric_quotient(collection):
+    """
+    The Isoperimetric quotient, defined as the ratio of a chain's area to the
+    area of the equi-perimeter circle.
+
+    Altman's PA_1 measure
+
+    Construction:
+    --------------
+    let:
+    p_d = perimeter of district
+    a_d = area of district
+
+    a_c = area of the constructed circle
+    r = radius of constructed circle
+
+    then the relationship between the constructed radius and the district
+    perimeter is:
+    p_d = 2 \pi r
+    p_d / (2 \pi) = r
+
+    meaning the area of the circle can be expressed as:
+    a_c = \pi r^2
+    a_c = \pi (p_d / (2\pi))^2
+
+    implying finally that the IPQ is:
+
+    pp = (a_d) / (a_c) = (a_d) / ((p_d / (2*\pi))^2 * \pi) = (a_d) / (p_d**2 / (4\PI))
+    """
+    return (4 * numpy.pi * pygeos.area(collection)) / (
+        pygeos.measurement.length(collection) ** 2
+    )
+
+
+def isoareal_quotient(collection):
+    """
+    The Isoareal quotient, defined as the ratio of a chain's perimeter to the
+    perimeter of the equi-areal circle
+
+    Altman's PA_3 measure, and proportional to the PA_4 measure
+    """
+    return (
+        2 * numpy.pi * numpy.sqrt(pygeos.area(collection) / numpy.pi)
+    ) / pygeos.measurement.length(collection)
+
+
+def minimum_bounding_circle_ratio(collection):
+    """
+    The Reock compactness measure, defined by the ratio of areas between the
+    minimum bounding/containing circle of a shape and the shape itself.
+
+    Measure A1 in Altman (1998), cited for Frolov (1974), but earlier from Reock
+    (1963)
+    """
+    mbc = pygeos.minimum_bounding_circle(collection)
+    return pygeos.area(collection) / pygeos.area(mbc)
+
+
+def radii_ratio(collection):
+    """
+    The Flaherty & Crumplin (1992) index, OS_3 in Altman (1998).
+
+    The ratio of the radius of the equi-areal circle to the radius of the MBC
+    """
+    r_eac = numpy.sqrt(pygeos.area(collection) / numpy.pi)
+    r_mbc = pygeos.minimum_bounding_radius(pointset)
```

Comment:
```suggestion
    r_mbc = pygeos.minimum_bounding_radius(collection)
```

`pointset` is not defined